gen2-road-segmentation/main.py

```diff
@@ -0,0 +1,150 @@
+    frame_colored = cv2.normalize(frame, None, 255, 0, cv2.NORM_INF, cv2.CV_8UC1)
+    frame_colored = cv2.equalizeHist(frame_colored)
+    frame_colored = cv2.applyColorMap(frame_colored, cv2.COLORMAP_HOT)
+    cv2.imshow(name, frame_colored)
+    return frame_colored
+
+
+class FPSHandler:
+    def __init__(self):
+        self.timestamp = time.time()
+        self.start = time.time()
+        self.frame_cnt = 0
+
+    def next_iter(self):
+        self.timestamp = time.time()
+        self.frame_cnt += 1
+
+    def fps(self):
+        return self.frame_cnt / (self.timestamp - self.start)
+
+
+def to_tensor_result(packet):
+    data = {}
+    for tensor in packet.getRaw().tensors:
+        if tensor.dataType == dai.TensorInfo.DataType.INT:
+            data[tensor.name] = np.array(packet.getLayerInt32(tensor.name)).reshape(tensor.dims)
+        elif tensor.dataType == dai.TensorInfo.DataType.FP16:
+            data[tensor.name] = np.array(packet.getLayerFp16(tensor.name)).reshape(tensor.dims)
+        elif tensor.dataType == dai.TensorInfo.DataType.I8:
+            data[tensor.name] = np.array(packet.getLayerUInt8(tensor.name)).reshape(tensor.dims)
+        else:
+            print("Unsupported tensor layer type: {}".format(tensor.dataType))
+    return data
+
+
+# Start defining a pipeline
+pipeline = dai.Pipeline()
+
+cam = pipeline.createColorCamera()
+cam.setResolution(dai.ColorCameraProperties.SensorResolution.THE_1080_P)
+cam.setBoardSocket(dai.CameraBoardSocket.RGB)
+
+# For deeplabv3
+cam.setColorOrder(dai.ColorCameraProperties.ColorOrder.BGR)
+cam.setPreviewSize(*nn_shape)
+cam.setInterleaved(False)
+
+# Define a neural network that will make predictions based on the source frames
+detection_nn = pipeline.createNeuralNetwork()
+detection_nn.setBlobPath(str(blobconverter.from_zoo(name='road-segmentation-adas-0001')))
+detection_nn.input.setBlocking(False)
+detection_nn.setNumInferenceThreads(2)
+cam.preview.link(detection_nn.input)
+
+# NN output linked to XLinkOut
+xout_nn = pipeline.createXLinkOut()
+xout_nn.setStreamName("nn")
+detection_nn.out.link(xout_nn.input)
+
+cam_xout = pipeline.createXLinkOut()
+cam_xout.setStreamName("cam")
+detection_nn.passthrough.link(cam_xout.input)
+
+# Left mono camera
+left = pipeline.createMonoCamera()
+left.setResolution(dai.MonoCameraProperties.SensorResolution.THE_400_P)
+left.setBoardSocket(dai.CameraBoardSocket.LEFT)
+# Right mono camera
+right = pipeline.createMonoCamera()
+right.setResolution(dai.MonoCameraProperties.SensorResolution.THE_400_P)
+right.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+
+# Create a node that will produce the depth map (using disparity output as it's easier to visualize depth this way)
+stereo = pipeline.createStereoDepth()
+stereo.initialConfig.setConfidenceThreshold(245)
+stereo.initialConfig.setMedianFilter(dai.StereoDepthProperties.MedianFilter.KERNEL_7x7)
+stereo.setLeftRightCheck(True)
+stereo.setDepthAlign(dai.CameraBoardSocket.RGB)
```

Comment:
use isp downscaling otherwise depth frame will be 1080P. Maybe remove depth frames completely? Or use cut-out depth based on semantic info, eg. only roads (as we have at gen2-deeplabv3_depth)?

Reply:
Good point, I removed the depth info for now to allow OAK-1 to run this example too. Later we can add depth crop and then even display as pointcloud with RGB alignment - but that shouldn't block this PR